uui-components/src/inputs/TextInput.tsx

```diff
@@ -78,8 +79,17 @@ export const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>((pro
         inputElement.current?.focus();
     };
 
-    const handleWrapperFocus = () => {
-        inputElement.current?.focus();
+    const handleWrapperFocus = (
+        event: React.FocusEvent<HTMLDivElement>,
+    ): void => {
+        /*
+            Without this check, when using Tab for navigation,
+            the focus always stays within the container, though it should move
+            to other elements (like clear button).
+        */
+        if (event.target === inputElement.current) {
```

Comment:
Potentially won't be necessary to make this check, if the focus management is simplified/refactored, removing focus/blur event handlers from the wrapper.